src/components/callback-notifications.tsx

```diff
@@ -0,0 +1,108 @@
+import { useEffect, useRef, useState } from "react"
+
+const duration = 3000
+interface NotificationType {
+  title: string
+  info: any
+  details: string
+}
+export const CallbackNotifications = ({ }) => {
+  const [notifications, setNotifications] = useState<(NotificationType | null)[]>([])
+  const notificationsRef = useRef<(NotificationType | null)[]>(notifications)
+
+  useEffect(() => {
+    const onMessage = (event: MessageEvent) => {
+      const origin = event.origin
+      // handle messages from the sandboxed block
+      const originRegex = new RegExp(
+        /^https:\/\/\d{1,4}-\d{1,4}-\d{1,4}-sandpack\.codesandbox\.io$/
+      );
+      if (!originRegex.test(origin) && origin !== window.location.origin) return;
+      const eventType = event.data.type || ""
+      // @ts-ignore
+      const type = {
+        "update-metadata": {
+          title: "Update metadata",
+          info: <>
+            New metadata:
+          </>,
+          details: JSON.stringify(event.data.metadata, null, 2),
+        },
+        "update-file": {
+          title: "Update file contents",
+          info: <>
+            New file contents:
+          </>,
+          details: JSON.stringify(event.data.content, null, 2),
+        },
+        "navigate-to-path": {
+          title: "Navigate to path",
+          info: <>
+            Requested to navigate to page:
+          </>,
+          details: event.data.path,
+        },
+        "github-data--request": {
+          title: "GitHub data request",
+          info: <>
+            Requested more data from GitHub: {event.data.requestType}, with config:
+          </>,
+          details: JSON.stringify(event.data.config, null, 2),
+        },
+      }[eventType] as NotificationType | undefined
```

Comment:
```
const type: NotificationType | undefined = { ...
```
instead of `as` will check that the expression has that type rather than casting it to that type.